### ISSUE TITLE
White screen after splash screen (iOS)

### ISSUE THREAD

**Issue by mluis** [apache/cordova-plugin-splashscreen#190]
# Bug Report

## Problem
After installing version 5.0.2 I get a white screen right after the splash screen.

### What is expected to happen?
Go to the login page after the splash screen.

### What does actually happen?
The splash screen disappears but the screen remains white.

## Information

These are the configuration options on config.xml.

```
    <preference name="AutoHideSplashScreen" value="true" />
    <preference name="SplashScreenDelay" value="3000" />
    <preference name="SplashScreen" value="screen" />
    <preference name="SplashShowOnlyFirstTime" value="false" />
```

### Environment, Platform, Device
Platform: iOS 9.3.5

### Version information
```
[WARN] Bad integration name: gulp

Ionic:

   ionic (Ionic CLI) : 4.10.3 (/Users/miguelluis/.nvm/versions/node/v11.3.0/lib/node_modules/ionic)
   Ionic Framework   : ionic1 1.1.1
   @ionic/v1-toolkit : 1.0.22

Cordova:

   cordova (Cordova CLI) : 9.0.0
   Cordova Platforms     : android 8.0.0, ios 5.0.0
   Cordova Plugins       : no whitelisted plugins (10 plugins total)

System:

   ios-deploy : 1.9.2
   ios-sim    : 6.1.3
   NodeJS     : v11.3.0 (/Users/miguelluis/.nvm/versions/node/v11.3.0/bin/node)
   npm        : 6.4.1
   OS         : macOS High Sierra
   Xcode      : Xcode 10.1 Build version 10B61
```

## Checklist

- [x] I searched for existing GitHub issues
- [x] I updated all Cordova tooling to most recent version
- [x] I included all the necessary information above

**Comment by pedrovitor074** [comment]
did u find some way to fix this?

**Comment by janpio** [comment]
Does it stay on white @mluis forever? Then your app is probably broken in some way.
Does it go to the app after it has finished loading? Then the correct way is probably to disable the autohide and manually call the `hide()` method when your app is ready. (And then optimize your Ionic app to load quicker so this doens't take so long.)

**Comment by IndieSW** [comment]
Have you tried using a splash screen theme?  And making the splash screen theme the same theme as the app theme?

**Comment by mluis** [comment]
It stays white forever. I also tried to call hide() manually but didn't worked.

It happened after updating the plugin although the same plugin version is working fine on mobile phone version iOS 12.2 but not on the iPad.

**Comment by janpio** [comment]
Persistent white screen is not really something that should be able to be caused by this plugin, strange. Have you remote debugged your app to look at the console for errors? https://ionic.zone/debug/remote-debug-your-app#ios

Did you also update Ionic, Cordova or Cordova Android at the same time?
Can you reproduce the problem in a plain new Cordova app created with `cordova create`?

**Comment by mluis** [comment]
The issue is showing the splash screen and then autohiding it following a white screen.
I just ran the remote debug you linked but the console doesn't present any errors.

The versions I'm using are those from Version Information. I don't recall upgrading everything at a time.

I will try asap to check if the problem exists in a new created app with cordova create.

**Comment by john-doherty** [comment]
I downgraded to 5.0.0 and it resolved the issue:

```bash
cordova plugin rm cordova-plugin-splashscreen --force
cordova plugin add cordova-plugin-splashscreen@5.0.0
```

**Comment by janpio** [comment]
The differences for iOS between 5.0.0 and 5.0.2 are minimal, so this doesn't really make sense @john-doherty. 
Is this reproducible in a new, plain Cordova app? A minimal reproduction repository would really help to debug and later fix this issue. More information on how to create one: https://github.com/apache/cordova-contribute/blob/master/create-reproduction.md

**Comment by Mapiac** [comment]
Any updates here? Same issues and versions

**Comment by EvelinJK** [comment]
Here with: 
```
cordova 10.0.0
cordova-ios 5.1.1
cordova-android 8.0.0
```

We had a similar issue with all newer versions of this plugin, but only after we installed `branch-cordova-sdk`
(beforehand it was working fine with `5.0.2`). 

**on iOS:** In our case we didn't have a white screen, but the splashscreen never disappeared
**on Android**: the app didn't break but the splashscreen didn't seem to work anymore (didn't appear)

This thread gave me the idea to test an older version – and after downgrading to `5.0.0` everything works again 🎊  (for some unknown reason).

The only hint we got came from the XCode console: 
With version `5.0.2` of this plugin it showed two errors: `Start Page at 'www/index.html' was not found` and `Internal navigation rejected - <allow-navigation> not set for url='about:blank'`
After downgrading to `5.0.0` those errors disappeared and the splash screen worked again. (Again, no clue why..).

**As always with cordova:** the combination of various plugins can yield ominous errors that make no sense at all 🤷‍♀️   – At least I'm too lazy to investigate further...

**Comment by breautek** [comment]
> Here with:
> 
> ```
> cordova 10.0.0
> cordova-ios 5.1.1
> cordova-android 8.0.0
> ```
> 
> We had a similar issue with all newer versions of this plugin, but only after we installed `branch-cordova-sdk`
> (beforehand it was working fine with `5.0.2`).
> 
> **on iOS:** In our case we didn't have a white screen, but the splashscreen never disappeared
> **on Android**: the app didn't break but the splashscreen didn't seem to work anymore
> 
> This thread gave me the idea to test an older version – and after downgrading to `5.0.0` everything works again  (for some unknown reason).
> 
> The only hint we got came from the XCode console:
> With version `5.0.2` of this plugin it showed two errors: `Start Page at 'www/index.html' was not found` and `Internal navigation rejected - <allow-navigation> not set for url='about:blank'`
> After downgrading to `5.0.0` those errors disappeared and the splash screen worked again. (Again, no clue why..).
> 
> **As always with cordova:** the combination of various plugins can yield ominous errors that make no sense at all  – At least I'm too lazy to investigate further...

If it's broken as of 5.0.2 but resolved not broken in 5.0.0, these are all the commits that has happened between those two tags.

https://github.com/apache/cordova-plugin-splashscreen/compare/rel/5.0.0...rel/5.0.2

**Comment by kmatherne** [comment]
I am having the same problem and am bumping this in hopes someone will respond with a quick fix at least.

**Comment by liuyao552** [comment]
v6.0.0 ios?don't has ios

**Comment by matrixreal** [comment]
> using the latest version (6.0.0 or 6.0.1-dev) i got same problem ... a white screen after splash screen even if i use <preference name="AutoHideSplashScreen" value="false" /> in config.xml and call navigator.splashscreen.hide(); on device ready.

> using 5.0.0 fix the problem (no white screen).

**Comment by breautek** [comment]
Closing as stale.

iOS Splashscreen code has been migrated to the core platform. If issue persists on current versions of cordova-ios, please raise a new issue in the [cordova-ios](https://github.com/apache/cordova-ios/issues) repository.
